cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerStatus.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tethering;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Information about tethered peers that's returned by REST APIs.
+ */
+public class PeerStatus {
```

Comment:
This seems almost the same as `PeerInfo`, with the addition of `connectionStatus`. Can this one actually just contain a `PeerInfo` and a `connectionStatus` fields instead? Also, why we need the connectionStatus field? Isn't that kind of overlapped with tetheringStatus?

Reply:
Created a base class `PeerBase` - `PeerInfo` and `PeerStatus` now extend `PeerBase`. Only difference between them is that PeerInfo stores the last connection time, whereas PeerStatus stores the connectivity status (active/inactive).
JSON serialized PeerInfo is stored in the tethering store. PeerStatus is returned by the REST API - used to populate connectivity state in the UI. `connectionStatus` is derived from `lastConnectionTime` (i.e. is current timestamp - lastConnectionTime < timeout?).